pkg/activator/net/lb_policy.go

```diff
@@ -84,3 +85,34 @@ func firstAvailableLBPolicy(ctx context.Context, targets []*podTracker) (func(),
 	}
 	return noop, nil
 }
+
+type roundRobinPolicyT struct {
+	mu  sync.Mutex
+	idx int
+}
+
+func newRoundRobinPolicy() lbPolicy {
+	rrp := roundRobinPolicyT{}
+	return func(ctx context.Context, targets []*podTracker) (func(), *podTracker) {
+		rrp.mu.Lock()
+		defer rrp.mu.Unlock()
```

Comment:
Can we get away without locking? We could try to get away using atomics potentially, though the benchmarks don't really warrant that I suppose.

Reply:
I tried, I was not happy with provable semantics for parallel requests.

Reply:
Fair enough, we can iterate if necessary :shrug:

Reply:
the problem, is that we start moving indices either independently or in interleaving fashion. In theory with enough requests it will still average out, but it's much harder to reason about